src/main/java/org/opentripplanner/gtfs/mapping/TripMapper.java

```diff
@@ -1,20 +1,24 @@
 package org.opentripplanner.gtfs.mapping;
 
+import java.time.Duration;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 import org.opentripplanner.framework.collection.MapUtils;
 import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.routing.api.request.framework.TimePenalty;
 import org.opentripplanner.transit.model.timetable.Trip;
 
 /** Responsible for mapping GTFS TripMapper into the OTP model. */
 class TripMapper {
 
   private final RouteMapper routeMapper;
   private final DirectionMapper directionMapper;
-  private TranslationHelper translationHelper;
+  private final TranslationHelper translationHelper;
 
   private final Map<org.onebusaway.gtfs.model.Trip, Trip> mappedTrips = new HashMap<>();
+  private final Map<Trip, TimePenalty> flexSafeDurationModifiers = new HashMap<>();
```

Comment:
The Sandbox model require you to make extentions and implement them in the sandbox, not directly inject code into existing classes. In this case we can make a separate FlexSafeDurationMapper in the sandbox. But making a proper extention is a bit difficult without making a mess. So, I think maybe we let this pass and spend time on cleaning up the internal model with respect to Flex features.

Reply:
Thanks.